components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/ListCmd.java

```diff
@@ -1,4 +1,113 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
 package org.wso2.apimgt.gateway.cli.cmd;
 
-public class ListCmd {
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.apimgt.gateway.cli.exception.CLIRuntimeException;
+import org.wso2.apimgt.gateway.cli.utils.GatewayCmdUtils;
+import org.wso2.apimgt.gateway.cli.utils.RouteUtils;
+import org.wso2.apimgt.gateway.cli.utils.SwaggerUtils;
+
+import java.io.PrintStream;
+import java.util.List;
+
+@Parameters(commandNames = "list", commandDescription = "list api/resources of the microgateway")
+public class ListCmd implements GatewayLauncherCmd{
+
+    @Parameter(hidden = true, required = true)
+    private List<String> mainArgs;
+
+    @Parameter(names = {"-ai", "--api-id"}, hidden = true)
```

Comment:
Since we are referring APIs and Resources only with and id shall we change this `--api`
```suggestion
    @Parameter(names = {"-a", "--api"}, hidden = true)
```

Reply:
-a flag is used in setup command to provide api name. Wouldn't it be a problem?

Reply:
That should be fine

Reply:
added the suggestion.